### PR TITLE
Dark Mode - NxGlobalSidebar - Interaction State Fixes - RSC-1677

### PR DESCRIPTION
**Pull request by avnvbhatta**
https://issues.sonatype.org/browse/RSC-1677

### CODE REVIEW

**Comment by EmilyBermudez88** [comment]
I created a ticket to expand the toggle button testing in [RSC-1676](https://issues.sonatype.org/browse/RSC-1676). I think the testing is related to this ticket (and close out RSC-1676), but should it be done here or in RSC-1676?

**Comment by EmilyBermudez88** [comment]
I think it may also be worth waiting for clarification on the sidebar's background color as well, as Ross brought up here: https://sonatype.slack.com/archives/CJUF9G7T2/p1682970512758289
I hadn't put it in the ticket yet because we were still waiting on design to confirm the background color choice.

**Comment by avnvbhatta** [comment]
> I created a ticket to expand the toggle button testing in [RSC-1676](https://issues.sonatype.org/browse/RSC-1676). I think the testing is related to this ticket (and close out RSC-1676), but should it be done here or in RSC-1676?

I'd prefer to do it here, cause this PR is directly related to the buttons anyway.

**Comment by avnvbhatta** [comment]
> I think it may also be worth waiting for clarification on the sidebar's background color, as Ross brought up here: https://sonatype.slack.com/archives/CJUF9G7T2/p1682970512758289 I hadn't put it in the ticket yet because we were still waiting on design to confirm the background color choice.

Yeah, let's wait for Jason's feedback for the background.

**Comment by EmilyBermudez88** [comment]
> > I think it may also be worth waiting for clarification on the sidebar's background color, as Ross brought up here: https://sonatype.slack.com/archives/CJUF9G7T2/p1682970512758289 I hadn't put it in the ticket yet because we were still waiting on design to confirm the background color choice.
> 
> Yeah, let's wait for Jason's feedback for the background.

So this has now been decided in Slack: https://sonatype.slack.com/archives/CJUF9G7T2/p1683115459812239, and a separate ticket has been added [RSC-1684](https://issues.sonatype.org/browse/RSC-1684). Wait for that to be merged to update the snapshots again, which will now have a different background color

**Comment by avnvbhatta** [comment]
> > > I think it may also be worth waiting for clarification on the sidebar's background color, as Ross brought up here: https://sonatype.slack.com/archives/CJUF9G7T2/p1682970512758289 I hadn't put it in the ticket yet because we were still waiting on design to confirm the background color choice.
> > 
> > 
> > Yeah, let's wait for Jason's feedback for the background.
> 
> So this has now been decided in Slack: https://sonatype.slack.com/archives/CJUF9G7T2/p1683115459812239, and a separate ticket has been added [RSC-1684](https://issues.sonatype.org/browse/RSC-1684) and completed. So snapshots will need to be updated with the new background color.

Updated snapshots here: 95b6c6c951179b4da44cee441ae7d9e08d948d94

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx [hunk 1]
```diff
@@ -463,22 +463,22 @@ const CssVariablesPage = () => {
             <ColorDocRow colorVar="--nx-color-global-sidebar-toggle">
               The color of the icon within the <NxCode>NxGlobalSidebar</NxCode> toggle button.
             </ColorDocRow>
-            <ColorDocRow colorVar="--nx-color-global-sidebar-toggle-background-hover">
+            <ColorDocRow colorVar="--nx-color-global-sidebar-background-hover">
```
Comment (EmilyBermudez88) [inline gallery/src/styles/CssVariables/CssVariablesPage.tsx:466]: The description for the variable, it applies to more than just the toggle now correct?

**Review by EmilyBermudez88 — APPROVED** [review]
other than comment, LGTM

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx (hunk 1 above)
Comment (avnvbhatta) [reply gallery/src/styles/CssVariables/CssVariablesPage.tsx:466]: > The description for the variable, it applies to more than just the toggle now correct?

Ah yes, it's for both the toggle and nav links. Updated doc here: aecbb088625c87e17569544c80c8610f470702b2

**Pull request merged**